index.js

```diff
@@ -105,31 +120,75 @@ function initApp(callback) {
-        return;
-      }
+     var Compendium = require('./lib/model/compendium');
 
-      // TODO implement ...
-    });
-    
-    app.get('/api/v1/compedium/:id/substition', function (req, res) {
+    app.post('/api/v1/substitution', (req, res) => {
       res.setHeader('Content-Type', 'application/json');
-      
-      // query substitions for given id from database
-
-      // create response
-      var response = {};
-      res.send(response);
+      // if (!req.isAuthenticated()) {
+      //   res.status(401).send('{"error":"not authenticated"}');
+      //   return;
+      // }
+      // if (req.user.level < config.user.level.substitute) {
+      //   res.status(401).send('{"error":"not allowed"}');
+      //   return;
+      // }
+
+      // new random id
+      let newID = randomstring.generate(config.id_length);
+
+      // TODO: check if id exists
+      let passon = {
+        user: "0000-0003-2287-369X",  // TODO: req.user.id
+        id: newID,
+        metadata: {
+          substituted: true,
+          substitution: req.body
+        }
+      };
+      debug('[%s] Starting substitution of new compendium [base: "%s" - overlay: "%s"] ...', passon.id, passon.metadata.substitution.base, passon.metadata.substitution.overlay);
+      return controllers.substitute.getMetadata(passon)   // get metadata
+        .then(controllers.substitute.checkOverlayId)      // check ERC id of overlay ERC
+        .then(controllers.substitute.createFolder)         // create folder with id
+        .then(controllers.substitute.copyBaseFiles)       // copy base files into folder
+        .then(controllers.substitute.copyOverlayFiles)    // copy overlay files into folder
+        .then(controllers.substitute.createDockerImage)   // TODO: later not necessary
+        .then(controllers.substitute.startDockerContainer)  // run docker container with given image
+        .then(controllers.substitute.writeYaml)             // write docker run cmd to erc.yml
+        .then(controllers.substitute.saveToDB)             // save to DB
+        .then((passon) => {
+            debug('[%s] Finished substitution of new compendium.', passon.id);
+            // console.log(passon);
+            res.status(200).send({'id': passon.id});
+        })
+        .catch(err => {
+            debug('[%s] - Error during substitution \n %s', passon.id, JSON.stringify(err));
+
+            let status = 500;
+            if (err.status) {
+              status = err.status;
+            }
+            let msg = 'Internal error';
+            if (err.msg) {
+              msg = err.msg;
+            }
+            res.status(status).send(JSON.stringify({ err: msg }));
+        });
     });
 
-    app.listen(config.net.port, () => {
-      debug('substitution %s with API version %s waiting for requests on port %s',
+    // GET list of subtsitutions
+    app.get('/api/v1/substitutions', controllers.substitutions.view);
```

Comment:
This does not reflect the current state of the API draft, please use `../substitution` (singular).

Reply:
done